dm/syncer/binlogstream/binlog_locations.go

```diff
@@ -161,16 +146,14 @@ func (l *locationRecorder) setCurEndGTID(e *replication.BinlogEvent) {
 // - curEndLocation is tried to be updated in-place
 // - txnEndLocation is assigned to curEndLocation when `e` is the last event of a transaction.
 func (l *locationRecorder) update(e *replication.BinlogEvent) {
-	l.mu.Lock()
-	defer l.mu.Unlock()
+	for _, f := range l.preUpdate {
+		f()
+	}
+	l.preUpdate = nil
```

Comment:
Are you about to clean it up? Seems appending new entries to a `nil` slice is safe in golang 😂 but a bit elusive for non-golang users. Would you like to write some comments to clarify it (just a recommendation)?